features/storage/blockdevice/VerifyReadBlockDevice.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** \addtogroup storage */
+/** @{*/
+
+#include "VerifyReadBlockDevice.h"
+#include "platform/mbed_error.h"
+#include "MbedCRC.h"
+
+namespace mbed {
+
+VerifyReadBlockDevice::VerifyReadBlockDevice(BlockDevice *bd)
+    : _bd(bd)
+{
+    // Does nothing
+}
+
+VerifyReadBlockDevice::~VerifyReadBlockDevice()
+{
+    // Does nothing
+}
+
+int VerifyReadBlockDevice::init()
+{
+    return _bd->init();
+}
+
+int VerifyReadBlockDevice::deinit()
+{
+    return _bd->deinit();
+}
+
+int VerifyReadBlockDevice::sync()
+{
+    return _bd->sync();
+}
+
+int VerifyReadBlockDevice::read(void *buffer, bd_addr_t addr, bd_size_t size)
+{
+    int ret, num_reads = 0;
+    uint32_t crc, prev_crc;
+
+    while ((num_reads < 2) || (crc != prev_crc)) {
```

Comment:
Consider adding an upper limit to num_reads to prevent an endless loop In case of a problem that might cause (crc!=prev_crc) forever.. 
(not sure its possible but for example if SPI frequency is wrong you might get different byte values per each read..)

Reply:
Thanks - added.